### PR TITLE
Implement bool parsing, tests, code generation

### PR DESCRIPTION
**Pull request by chhaylatte**

### CODE REVIEW

File: OriginateTheme/Scripts/ot_generator.py [hunk 1]
```diff
@@ -151,6 +151,33 @@ def createColorGetter(color, name, keyPath):
...
+        Object of type 'Color'.
+        name: String
+        The name of the property to create a getter for.
+        keyPath: String
+        The name of the key path to associated to the property.
+        """
+    getter = """
+        - (UIColor *)$name
+        {
+        if (!_$name) {
```
Comment (allewun) [inline OriginateTheme/Scripts/ot_generator.py]: make sure to indent the generated code properly (following the conventions that are already being used)

Comment (allewun) [inline OriginateTheme/Scripts/ot_generator.py]: also, this looks like it's a copy-pasted UIColor getter, not a BOOL getter.

File: OriginateTheme/Scripts/ot_generator.py [hunk 2]
```diff
@@ -449,11 +518,12 @@ def generateComponentsClassHeaderFileContent(className, components):
...
+    createComponentPropertyName = lambda componentKey, key, uiType:  componentKey + upcaseFirstLetter(key) + upcaseFirstLetter(uiType)
     createPropertyDefinition = lambda componentKey, key, uiType, accessType: createProperty('strong', accessType, 'UI' + upcaseFirstLetter(uiType), createComponentPropertyName(componentKey, key, uiType))
+    createValuePropertyDefinition = lambda componentKey, key, objcType, accessType: createValueProperty(accessType, objcType, createComponentPropertyName(componentKey, key, ""))
 
     # Create the public properties.
-    OriginateThemePublicProperties = [[[createPropertyDefinition(component.key, c.key, 'color', 'readonly') for c in component.colors], [createPropertyDefinition(component.key, f.key, 'font', 'readonly') for f in component.fonts], ['']] for component in components]
+    OriginateThemePublicProperties = [[[createPropertyDefinition(component.key, c.key, 'color', 'readonly') for c in component.colors], [createPropertyDefinition(component.key, f.key, 'font', 'readonly') for f in component.fonts], [createValuePropertyDefinition(component.key, b.key, 'BOOL', 'readonly') for b in component.bools], ['']] for component in components]
```
Comment (allewun) [inline OriginateTheme/Scripts/ot_generator.py]: can we put each element of this array on a newline and align them? Maybe something like this?

```python
OriginateThemePublicProperties = [[
  [createPropertyDefinition(component.key, c.key, 'color', 'readonly') for c in component.colors],
  [createPropertyDefinition(component.key, f.key, 'font', 'readonly') for f in component.fonts],
  [createValuePropertyDefinition(component.key, b.key, 'BOOL', 'readonly') for b in component.bools],
  ['']
] for component in components]
```

File: OriginateTheme/Scripts/ot_generator.py (hunk 1 above)
Comment (chhaylatte) [inline OriginateTheme/Scripts/ot_generator.py]: Accidentally committed this =/

**Comment by chhaylatte** [comment]
Lots of updates.  Needs review.

File: OriginateTheme/Scripts/ot_generator.py [hunk 3]
```diff
@@ -151,6 +151,29 @@ def createColorGetter(color, name, keyPath):
 }"""
     return string.Template(getter).substitute({ 'name' : name, 'colorKeyPathKey' : keyPath, 'colorCode' : color.code})
 
+def createBoolGetter(bool, name, keyPath):
+    """
+        Creates a new getter to access a OriginateTheme UIColor property.
```
Comment (allewun) [inline OriginateTheme/Scripts/ot_generator.py]: update comment

File: OriginateTheme.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -109,6 +113,8 @@
 				649754601D8B1B1300A3B3D0 /* UIColor+OTKeySource.m */,
 				649754611D8B1B1300A3B3D0 /* UIFont+OTKeySource.h */,
 				649754621D8B1B1300A3B3D0 /* UIFont+OTKeySource.m */,
+				CE1AFCCF1DECDF570036657B /* NSValue+OTKeySource.h */,
```
Comment (allewun) [inline OriginateTheme.xcodeproj/project.pbxproj]: please sort alphabetically

File: OriginateTheme/Sources/Categories/NSValue+OTKeySource.h [hunk 5]
```diff
@@ -0,0 +1,16 @@
+//
+//  NSValue+OTKeySource.h
+//  OriginateTheme
+//
+//  Created by Danny Chhay on 11/28/16.
```
Comment (pkluz) [inline OriginateTheme/Sources/Categories/NSValue+OTKeySource.h]: Let's get this fixed as well - @chhaylatte do you mind doing me a favor and setting your date settings in macOS as follows:

System Preferences -> Language & Region -> Advanced -> Dates -> Short:

screen shot 2016-11-28 at 11 54 18 pm

**Comment by allewun** [comment]
LGTM, any other comments from @rweindl ?

**Comment by chhaylatte** [comment]
Do I wait for approval from more people or do I just merge it?

**Comment by allewun** [comment]
It looks fine to me, but I'm not a python guy so I figured it might be good to have @rweindl have the final word, since he created this initially too. It's not blocking anything at the moment, so merging ASAP might not be necessary.

**Comment by rweindl** [comment]
Hey looks also fine for me! Good job. :)

**Pull request merged**